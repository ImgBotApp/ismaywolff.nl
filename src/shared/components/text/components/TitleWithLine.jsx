import React from 'react'
import styled from 'styled-components'
import { bool, string, node, oneOf } from 'prop-types'

/* istanbul ignore next: is tested but not registered by istanbul */
const TitleWithLineTemplate = styled.h1`
  background: var(--color-white);
  display: inline;
  font-size: ${props => props.size};
  margin: ${props => props.margin};
  padding: 0 1rem;
`

const Wrapper = styled.div`
  position: relative;
  text-align: center;

  &::after {
    border-bottom: var(--border-size) solid var(--color-gray);
    bottom: 0;
    content: "";
    left: 0;
    position: absolute;
    right: 0;
    top: 0;
    transform: translateY(-50%);
    z-index: -1;
  }
`

const TitleWithLineFactory = ({ tag, size, children, center, margin }) => {
  const TitleWithLine = TitleWithLineTemplate.withComponent(tag)
  return (
    <Wrapper>
      <TitleWithLine size={size} center={center} margin={margin}>
        {children}
      </TitleWithLine>
    </Wrapper>
  )
}

TitleWithLineFactory.propTypes = {
  center: bool,
  children: node,
  margin: string,
  size: string,
  tag: oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6']).isRequired
}

TitleWithLineFactory.defaultProps = {
  center: false,
  children: undefined,
  margin: 'unset',
  size: '1rem'
}

export default TitleWithLineFactory
