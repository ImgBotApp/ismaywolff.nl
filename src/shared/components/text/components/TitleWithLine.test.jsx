import 'jest-styled-components'
import React from 'react'
import { shallow } from 'enzyme'
import TitleWithLine from './TitleWithLine'

describe('<TitleWithLine />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<TitleWithLine tag="h1">Title</TitleWithLine>)
    expect(wrapper).toMatchSnapshot()
  })

  it('accepts a center property', () => {
    const wrapper = shallow(
      <TitleWithLine tag="h1" center>
        Title
      </TitleWithLine>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('accepts a margin property', () => {
    const wrapper = shallow(
      <TitleWithLine tag="h1" margin="1rem">
        Title
      </TitleWithLine>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('accepts a size property', () => {
    const wrapper = shallow(
      <TitleWithLine tag="h1" size="2rem">
        Title
      </TitleWithLine>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
