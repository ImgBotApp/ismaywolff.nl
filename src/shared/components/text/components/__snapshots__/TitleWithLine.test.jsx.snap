// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleWithLine /> accepts a center property 1`] = `
<styled.div>
  <styled.h1
    center={true}
    margin="unset"
    size="1rem"
  >
    Title
  </styled.h1>
</styled.div>
`;

exports[`<TitleWithLine /> accepts a margin property 1`] = `
<styled.div>
  <styled.h1
    center={false}
    margin="1rem"
    size="1rem"
  >
    Title
  </styled.h1>
</styled.div>
`;

exports[`<TitleWithLine /> accepts a size property 1`] = `
<styled.div>
  <styled.h1
    center={false}
    margin="unset"
    size="2rem"
  >
    Title
  </styled.h1>
</styled.div>
`;

exports[`<TitleWithLine /> renders correctly 1`] = `
<styled.div>
  <styled.h1
    center={false}
    margin="unset"
    size="1rem"
  >
    Title
  </styled.h1>
</styled.div>
`;
